applications/development: /s/name/pname&version/

Stunkymonkey opened the pull request:
###### Motivation for this change
#103997

###### Things done
partly automated renaming of `name` to `pname` & `version`.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented:
Please keep those PRs to a reviewable size. I am comfortable reviewing and merging 80 to 100 changes at the time if they cause no eval errors.

SuperSandro2000 commented on pkgs/development/mobile/flashtool/default.nix:
```diff
@@ -12,11 +12,12 @@
+    name = "${pname}-${version}-linux.tar.7z";
     sha256 = "0mfjdjj7clz2dhkg7lzy1m8hk8ngla7zgcryf51aki1gnpbb2zc1";
   };
 
```
Please update the tar xf to command to ``tar xf ${pname}-${version}-linux.tar``

SuperSandro2000 commented on pkgs/development/tools/xib2nib/default.nix:
```diff
@@ -1,7 +1,8 @@
 stdenv.mkDerivation {
-  name = "xib2nib-730e177";
+  pname = "xib2nib";
+  version = "730e177";
```
```suggestion
  version = "unstable-2017-04-12";
```

SuperSandro2000 commented on pkgs/development/tools/wiggle/default.nix:
```diff
@@ -1,11 +1,12 @@
 
   src = fetchurl {
-    url = "https://github.com/neilbrown/wiggle/archive/v1.3.tar.gz";
+    url = "https://github.com/neilbrown/${pname}/archive/v${version}.tar.gz";
```
Let me fix that in another PR and use fetchFromGitHub.

SuperSandro2000 commented on pkgs/development/tools/sunxi-tools/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitHub {
     owner = "linux-sunxi";
-    repo = "sunxi-tools";
+    repo = ${pname};
```
I am not a big fan of this because technically the repo name does not need to match the pname but it does often.

SuperSandro2000 commented on pkgs/development/tools/sunxi-tools/default.nix:
```diff
@@ -1,11 +1,12 @@
-  name = "sunxi-tools-20181113";
+stdenv.mkDerivation rec {
+  pname = "sunxi-tools";
+  version = "20181113";
```
```suggestion
  version = "unstable-2018-11-14";
```

SuperSandro2000 commented on pkgs/development/tools/ocaml/ocamlscript/default.nix:
```diff
@@ -1,8 +1,9 @@
+  version = "2.0.3";
   src = fetchurl {
-    url = "http://mjambon.com/releases/ocamlscript/ocamlscript-2.0.3.tar.gz";
+    url = "http://mjambon.com/releases/${pname}/${pname}-${version}.tar.gz";
```
```suggestion
    url = "https://mjambon.com/releases/${pname}/${pname}-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/development/tools/ocaml/ocamlify/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://forge.ocamlcore.org/frs/download.php/1209/ocamlify-0.0.2.tar.gz";
+    url = "http://forge.ocamlcore.org/frs/download.php/1209/${pname}-${version}.tar.gz";
```
```suggestion
    url = "https://forge.ocamlcore.org/frs/download.php/1209/${pname}-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/development/tools/neoload/default.nix:
```diff
@@ -30,15 +30,16 @@ let dotInstall4j = path: writeTextFile { name = "dot-install4j"; text = ''
   src = fetchurl (
     if stdenv.hostPlatform.system == "x86_64-linux" then
-      { url = "http://neoload.installers.neotys.com/documents/download/neoload/v4.1/neoload_4_1_4_linux_x64.sh";
+      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v4.1/${pname}_4_1_4_linux_x64.sh";
```
```suggestion
      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_4_1_4_linux_x64.sh";
```

SuperSandro2000 commented on pkgs/development/tools/neoload/default.nix:
```diff
@@ -30,15 +30,16 @@ let dotInstall4j = path: writeTextFile { name = "dot-install4j"; text = ''
         sha256 = "199jcf5a0nwfm8wfld2rcjgq64g91vvz2bkmki8dxfzf1yasifcd"; }
     else
-      { url = "http://neoload.installers.neotys.com/documents/download/neoload/v4.1/neoload_4_1_4_linux_x86.sh";
+      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v4.1/${pname}_4_1_4_linux_x86.sh";
```
```suggestion
      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_4_1_4_linux_x86.sh";
```

SuperSandro2000 commented on pkgs/development/tools/misc/uisp/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://mirror.switch.ch/mirror/gentoo/distfiles/uisp-20050207.tar.gz";
+    url = "http://mirror.switch.ch/mirror/gentoo/distfiles/${pname}-${version}.tar.gz";
```
```suggestion
    url = "https://mirror.switch.ch/mirror/gentoo/distfiles/${pname}-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/development/tools/misc/gnum4/default.nix:
```diff
@@ -6,7 +6,8 @@
+  version = "1.4.18";
 
   src = fetchurl {
     url = "mirror://gnu/m4/m4-1.4.18.tar.bz2";
```
```suggestion
    url = "mirror://gnu/m4/m4-${version}.tar.bz2";
```
this is my preferred style to only substitute the version number.

SuperSandro2000 commented on pkgs/development/tools/misc/gnum4/default.nix:
```diff
@@ -6,7 +6,8 @@
 # files.
 
 stdenv.mkDerivation {
```
```suggestion
stdenv.mkDerivation rec {
```

SuperSandro2000 commented on pkgs/development/tools/misc/drush/default.nix:
```diff
@@ -11,8 +12,10 @@ stdenv.mkDerivation rec {
-    url    = "https://github.com/drush-ops/drush/archive/6.1.0.tar.gz";
+  src = fetchFromGitHub {
+    owner = "drush-ops";
+    repo  = "${pname}";
+    rev = "${version}";
```
```suggestion
    repo  = pname;
    rev = version;
```

SuperSandro2000 commented on pkgs/development/tools/misc/autobuild/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://savannah.spinellicreations.com/autobuild/${name}.tar.gz";
+    url = "http://savannah.spinellicreations.com/${pname}/${pname}-${version}.tar.gz";
```
Name is not resoling. Can we use any other savannah mirror here or the mirror:// style url?

SuperSandro2000 commented on pkgs/development/tools/analysis/valkyrie/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://valgrind.org/downloads/${name}.tar.bz2";
+    url = "http://valgrind.org/downloads/${pname}-${version}.tar.bz2";
```
```suggestion
    url = "https://valgrind.org/downloads/${pname}-${version}.tar.bz2";
```

SuperSandro2000 commented on pkgs/development/tools/analysis/swarm/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitHub {
     owner = "nimble-code";
-    repo = "swarm";
+    repo = ${pname};
```
```suggestion
    repo = pname;
```

SuperSandro2000 commented on pkgs/development/tools/analysis/swarm/default.nix:
```diff
@@ -1,11 +1,12 @@
-  name = "swarm-2019-03-11";
+stdenv.mkDerivation rec {
+  pname = "swarm";
+  version = "2019-03-11";
```
```suggestion
  version = "unstable-2019-03-11";
```

SuperSandro2000 commented on pkgs/development/tools/analysis/splint/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://www.splint.org/downloads/${name}.src.tgz";
+    url = "http://www.splint.org/downloads/${pname}-${version}.src.tgz";
```
```suggestion
    url = "https://www.splint.org/downloads/${pname}-${version}.src.tgz";
```

SuperSandro2000 commented on pkgs/development/tools/analysis/makefile2graph/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitHub {
     owner = "lindenb";
-    repo = "makefile2graph";
+    repo = ${pname};
```
```suggestion
    repo = pname;
```

SuperSandro2000 commented on pkgs/development/tools/analysis/makefile2graph/default.nix:
```diff
@@ -1,11 +1,12 @@
-  name = "makefile2graph-2018-01-03";
+stdenv.mkDerivation rec {
+  pname = "makefile2graph";
+  version = "2018-01-03";
```
```suggestion
  version = "unstable-2018-01-03";
```

SuperSandro2000 commented on pkgs/development/tools/analysis/garcosim/tracefilesim/default.nix:
```diff
@@ -2,7 +2,8 @@
 
-  name = "tracefilesim-2015-11-07";
+  pname = "tracefilesim";
+  version = "2015-11-07";
```
```suggestion
  version = "unstable-2015-11-07";
```

SuperSandro2000 commented on pkgs/development/tools/analysis/garcosim/tracefilesim/default.nix:
```diff
@@ -2,7 +2,8 @@
+  pname = "tracefilesim";
+  version = "2015-11-07";
 
   src = fetchgit {
```
This should be fetchFromGitHub.

SuperSandro2000 commented on pkgs/development/tools/analysis/garcosim/tracefilegen/default.nix:
```diff
@@ -2,7 +2,8 @@
 
-  name = "tracefilegen-2017-05-13";
+  pname = "tracefilegen";
+  version = "2017-05-13";
```
```suggestion
  version = "unstable-2017-05-13";
```

SuperSandro2000 commented on pkgs/development/python-modules/gprof2dot/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitHub {
     owner = "jrfonseca";
-    repo = "gprof2dot";
+    repo = ${pname};
```
```suggestion
    repo = pname;
```

SuperSandro2000 commented on pkgs/development/python-modules/gprof2dot/default.nix:
```diff
@@ -1,11 +1,12 @@
     owner = "jrfonseca";
-    repo = "gprof2dot";
+    repo = ${pname};
     rev = "2019.11.30";
```
```suggestion
    rev = version;
```

SuperSandro2000 commented on pkgs/development/python-modules/gprof2dot/default.nix:
```diff
@@ -1,11 +1,12 @@
 { lib, fetchFromGitHub, buildPythonApplication, python, graphviz }:
 
 buildPythonApplication {
```
```suggestion
buildPythonApplication rec {
```

SuperSandro2000 commented on pkgs/development/mobile/flashtool/default.nix:
```diff
@@ -12,11 +12,12 @@
+  version = "0.9.14.0";
 
   src = requireFile {
     url = "http://dfiles.eu/files/n8c1c3pgc";
```
```suggestion
    url = "https://dfiles.eu/files/n8c1c3pgc";
```

~~That site looks very questionable. I would be for droping this package.~~

```
This file does not exist, the access to the following file is limited or it has been removed due to infringement of copyright. 
```

Lets remove it.

SuperSandro2000 commented on pkgs/build-support/libredirect/default.nix:
```diff
@@ -1,14 +1,15 @@
   '';
 
-  libName = "libredirect" + stdenv.targetPlatform.extensions.sharedLibrary;
+  libName = "${pname}" + stdenv.targetPlatform.extensions.sharedLibrary;
```
```suggestion
  libName = "${pname}${stdenv.targetPlatform.extensions.sharedLibrary}";
```

SuperSandro2000 commented on pkgs/applications/audio/qmmp/default.nix:
```diff
@@ -29,10 +29,11 @@
 
   src = fetchurl {
-    url = "http://qmmp.ylsoftware.com/files/${name}.tar.bz2";
+    url = "http://qmmp.ylsoftware.com/files/${pname}-${version}.tar.bz2";
```
```suggestion
    url = "https://qmmp.ylsoftware.com/files/${pname}-${version}.tar.bz2";
```

SuperSandro2000 commented on pkgs/applications/audio/opus-tools/default.nix:
```diff
@@ -1,9 +1,11 @@
+
   src = fetchurl {
-    url = "http://downloads.xiph.org/releases/opus/${name}.tar.gz";
+    url = "http://downloads.xiph.org/releases/opus/${pname}-${version}.tar.gz";
```
```suggestion
    url = "https://downloads.xiph.org/releases/opus/${pname}-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/applications/audio/mp3gain/default.nix:
```diff
@@ -1,9 +1,10 @@
+  version = "1.6.2";
   src = fetchurl {
-    url = "mirror://sourceforge/mp3gain/mp3gain-1_6_2-src.zip";
+    url = "mirror://sourceforge/${pname}/${pname}-1_6_2-src.zip";
```
```suggestion
    url = "mirror://sourceforge/${pname}/${pname}-${lib.replaceStrings ["."] ["_"] version}-src.zip";
```

SuperSandro2000 commented on pkgs/applications/audio/gmu/default.nix:
```diff
@@ -5,10 +5,11 @@
 
   src = fetchurl {
-    url = "http://wejp.k.vu/files/${name}.tar.gz";
+    url = "http://wejp.k.vu/files/${pname}-${version}.tar.gz";
```
```suggestion
    url = "https://wej.k.vu/files/${pname}-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/applications/audio/gbsplay/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitHub {
     owner = "mmitch";
-    repo = "gbsplay";
+    repo = ${pname};
```
```suggestion
    repo = pname;
```

SuperSandro2000 commented on pkgs/applications/audio/gbsplay/default.nix:
```diff
@@ -1,11 +1,12 @@
-  name = "gbsplay-2016-12-17";
+stdenv.mkDerivation rec {
+  pname = "gbsplay";
+  version = "2016-12-17";
```
```suggestion
  version = "unstable-2016-12-17";
```

SuperSandro2000 commented on pkgs/applications/audio/dirt/default.nix:
```diff
@@ -1,7 +1,8 @@
 stdenv.mkDerivation {
-  name = "dirt-2018-01-01";
+  pname = "dirt";
+  version = "2018-01-01";
```
```suggestion
  version = "unstable-2018-01-01";
```

SuperSandro2000 commented on pkgs/applications/audio/cdparanoia/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://downloads.xiph.org/releases/cdparanoia/${name}.src.tgz";
+    url = "http://downloads.xiph.org/releases/${pname}/${pname}-${version}.src.tgz";
```
```suggestion
    url = "https://downloads.xiph.org/releases/${pname}/${pname}-${version}.src.tgz";
```

SuperSandro2000 commented on pkgs/applications/audio/aacgain/default.nix:
```diff
@@ -1,11 +1,12 @@
   src = fetchFromGitLab {
     owner = "mulx";
-    repo = "aacgain";
+    repo = "${pname}";
```
```suggestion
    repo = pname;
```

Stunkymonkey committed:
flashtool: remove

Because the download is down for copyright reasons and the site looks very fishy and very similar to the ones used to distribute content illegally.

Stunkymonkey committed:
tracefilesim: replace fetchgit with fetchFromGitHub

Stunkymonkey committed:
drush: use fetchFromGitHub

Stunkymonkey committed:
wiggle: use fetchFromGitHub with pname and version

SuperSandro2000 commented on pkgs/development/tools/misc/drush/default.nix:
```diff
@@ -11,8 +12,10 @@ stdenv.mkDerivation rec {
+    owner = "drush-ops";
+    repo  = pname;
+    rev = version;
     sha256 = "1jgnc4jjyapyn04iczvcz92ic0vq8d1w8xi55ismqyy5cxhqj6bp";
```
I am not sure if this hash and others are still the same after switching to fetchFromGitHub.

Ma27 replied:
I'm pretty sure that this isn't the case for at least fetchgit since the first one does a `git clone` while `fetchFromGitHub` fetches a tarball from GitHub's API.

Stunkymonkey replied:
i am currently compiling. so i will see what is the outcome

talyz replied:
Yep, the hash should be different, since `fetchFromGitHub` unpacks the contents of the archive.

SuperSandro2000 replied:
```suggestion
    sha256 = "sha256-0nf/m+xJmfHsFLuordiMp8UyrGGXuS70+zFHkIxLWhU=";
```

Stunkymonkey replied:
i compiled it successfully with the given sha256sum

SuperSandro2000 commented on pkgs/development/tools/neoload/default.nix:
```diff
@@ -30,15 +30,16 @@ let dotInstall4j = path: writeTextFile { name = "dot-install4j"; text = ''
   src = fetchurl (
     if stdenv.hostPlatform.system == "x86_64-linux" then
-      { url = "http://neoload.installers.neotys.com/documents/download/neoload/v4.1/neoload_4_1_4_linux_x64.sh";
+      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_4_1_4_linux_x64.sh";
```
```suggestion
      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_${replaceStrings [ "." ] [ "_" ] version}_linux_x64.sh";
```

SuperSandro2000 replied:
I think you accidentally reverted this.

SuperSandro2000 commented on pkgs/development/tools/neoload/default.nix:
```diff
@@ -30,15 +30,16 @@ let dotInstall4j = path: writeTextFile { name = "dot-install4j"; text = ''
         sha256 = "199jcf5a0nwfm8wfld2rcjgq64g91vvz2bkmki8dxfzf1yasifcd"; }
     else
-      { url = "http://neoload.installers.neotys.com/documents/download/neoload/v4.1/neoload_4_1_4_linux_x86.sh";
+      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_4_1_4_linux_x86.sh";
```
```suggestion
      { url = "http://neoload.installers.neotys.com/documents/download/${pname}/v${lib.versions.majorMinor version}/${pname}_${replaceStrings [ "." ] [ "_" ] version}_linux_x86.sh";
```

SuperSandro2000 commented on pkgs/development/tools/wiggle/default.nix:
```diff
@@ -1,12 +1,15 @@
-{ lib, stdenv, fetchurl, ncurses, groff }:
```
I already fixed this package in https://github.com/NixOS/nixpkgs/commit/2b6b345900f470dde8bdbeea24fddf9ff37749ed. It would be great if you could drop the changes from this PR to prevent a merge conflict.

SuperSandro2000 commented on pkgs/development/tools/misc/autoconf/default.nix:
```diff
@@ -6,10 +6,11 @@
 
 stdenv.mkDerivation rec {
-  name = "autoconf-2.71";
+  pname = "autoconf";
```
I think we should move this one change into a PR that targets staging. I have the hope that all other rebuilds combined are below 500 packages so we can merge it directly into master.

Stunkymonkey replied:
so the main goal is to only have "small" changes, that have big impact merged on `staging`? and try to have the rest merged into `master`

jtojnar commented on pkgs/applications/audio/cdparanoia/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://downloads.xiph.org/releases/cdparanoia/${name}.src.tgz";
+    url = "https://downloads.xiph.org/releases/${pname}/${pname}-${version}.src.tgz";
```
The directory name does not exist.

SuperSandro2000 replied:
https://downloads.xiph.org/releases/cdparanoia/cdparanoia-III-10.2.src.tgz WOFM

jtojnar replied:
This link resolves to `downloads.xiph.org/releases/cdparanoia-III/cdparanoia-III-10.2.src.tgz`, though.

stigtsp commented on pkgs/top-level/all-packages.nix:
```diff
@@ -4725,10 +4725,6 @@ in
 
   flamerobin = callPackage ../applications/misc/flamerobin { };
 
-  flashtool = pkgsi686Linux.callPackage ../development/mobile/flashtool {
-    inherit (androidenv.androidPkgs_9_0) platform-tools;
-  };
-
```
Why is `flashtool` removed? Can't find a reason mentioned in the PR or commit

Stunkymonkey replied:
@SuperSandro2000 suggested: https://github.com/NixOS/nixpkgs/pull/126894#discussion_r651379506

SuperSandro2000 replied:
Because the download is down and the site looks very fishy and very similar to the ones used to distribute content illegally.

stigtsp replied:
Fair enough. But the reason for this should be noted in the commit (and PR description) imho.

jtojnar commented on pkgs/development/tools/ocaml/camlp5/default.nix:
```diff
@@ -4,12 +4,13 @@ if lib.versionOlder ocaml.version "4.02"
+  pname = "camlp5";
+  version = "7.14";
 
   src = fetchzip {
```
This can be replaced by `fetchFromGitHub`.

Stunkymonkey replied:
@SuperSandro2000 suggested:

https://github.com/NixOS/nixpkgs/pull/126894#discussion_r651379506

jtojnar commented on pkgs/development/tools/misc/uisp/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://mirror.switch.ch/mirror/gentoo/distfiles/uisp-20050207.tar.gz";
+    url = "https://mirror.switch.ch/mirror/gentoo/distfiles/${pname}-${version}.tar.gz";
```
This link seems to be dead.

jtojnar commented on pkgs/build-support/libredirect/default.nix:
```diff
@@ -1,14 +1,15 @@
   '';
 
-  libName = "libredirect" + stdenv.targetPlatform.extensions.sharedLibrary;
+  libName = "${pname}${stdenv.targetPlatform.extensions.sharedLibrary}";
```
This probably should not depend on `pname`. Package gets name from the library, not the other way around.

SuperSandro2000 commented:
Anyone has an idea what is causing the mass rebuild here? I would like to move those packages to a PR which targets staging if we can.

Stunkymonkey committed:
camlp5: replace fetchzip with fetchFromGitHub

Stunkymonkey committed:
application/development: /s/name/pname&version/

SuperSandro2000 commented on pkgs/development/tools/wiggle/default.nix:
```diff
@@ -1,14 +1,15 @@
 
 stdenv.mkDerivation rec {
-  name = "wiggle";
+
```
```suggestion
```

Stunkymonkey commented:
i know, that this has many rebuilds, but is there anything more to do?

SuperSandro2000 commented on pkgs/development/tools/misc/autobuild/default.nix:
```diff
@@ -1,10 +1,11 @@
 
   src = fetchurl {
-    url = "http://savannah.spinellicreations.com/autobuild/${name}.tar.gz";
+    url = "mirror://savannah.spinellicreations.com/${pname}/${pname}-${version}.tar.gz";
```
```suggestion
    url = "mirror://savannah/${pname}/${pname}-${version}.tar.gz";
```

SuperSandro2000 reviewed (changes requested):
except that LGTM
